src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -777,198 +1120,238 @@ s32 func_80A12B78(EnGo* this, PlayState* play) {
     return 0;
 }
 
-s32 func_80A12C48(EnGo* this, PlayState* play, s32 arg2) {
+/**
+ * Change the Goron's animation.
+ *
+ * Goron animations come from one of three categories
+ * - Basic Goron animations
+ * - Goron Gymnastics like Stretches, Cheers, etc... (Taisou)
+ * - Spring has Arrived Cutscene Animationstodo (Hakugin)
+ * Changing animations with this function handles the transitions between the three categories.
+ *
+ * @return true if animation request was valid
+ */
+s32 EnGo_ChangeAnimation(EnGo* this, PlayState* play, EnGoAnimationIndex anim) {
     s8 objIdx = this->actor.objBankIndex;
     s8 objIdx2 = -1;
     s32 ret = false;
 
-    if ((arg2 >= 18) && (this->unk_289 >= 0)) {
-        objIdx2 = this->unk_289;
-    } else if ((arg2 >= 10) && (this->unk_288 >= 0)) {
-        objIdx2 = this->unk_288;
-    } else if (arg2 < 10) {
+    if ((anim >= ENGO_ANIM_HAKUGIN_START) && (this->indexHakuginDemo >= 0)) {
+        objIdx2 = this->indexHakuginDemo;
+    } else if ((anim >= ENGO_ANIM_TAISOU_START) && (this->indexTaisou >= 0)) {
+        objIdx2 = this->indexTaisou;
+    } else if (anim < ENGO_ANIM_TAISOU_START) {
         objIdx2 = this->actor.objBankIndex;
     }
 
     if (objIdx2 >= 0) {
         gSegments[6] = VIRTUAL_TO_PHYSICAL(play->objectCtx.status[objIdx2].segment);
-        this->unk_3DC = arg2;
-        ret = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, arg2);
-        this->unk_398 = this->skelAnime.playSpeed;
+        this->currAnimIndex = anim;
+        ret = SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, anim);
+        this->currAnimPlaySpeed = this->skelAnime.playSpeed;
         gSegments[6] = VIRTUAL_TO_PHYSICAL(play->objectCtx.status[objIdx].segment);
     }
 
     return ret;
 }
 
-void func_80A12D6C(EnGo* this) {
-    if ((this->unk_390 & 0x20) && (DECR(this->unk_3BC) == 0)) {
-        this->unk_3BE++;
-        if (this->unk_3BE >= 4) {
-            this->unk_3BC = Rand_S16Offset(30, 30);
-            this->unk_3BE = 0;
+/**
+ * Update the eye textures to blink.
+ */
+void EnGo_UpdateEyes(EnGo* this) {
+    if (this->actionFlags & ENGO_FLAG_EYESOPEN) {
+        if (DECR(this->eyeTimer) == 0) {
+            this->eyeTexIndex++;
+            if (this->eyeTexIndex >= ENGO_EYETEX_CLOSED2) {
+                this->eyeTimer = Rand_S16Offset(30, 30);
+                this->eyeTexIndex = ENGO_EYETEX_OPEN;
+            }
         }
     }
 }
 
-void func_80A12DF4(EnGo* this, PlayState* play) {
-    if (this->unk_3D4 == 0) {
-        func_80A12C48(this, play, 9);
-        this->unk_3D4++;
-    } else if ((this->unk_3D4 == 1) && Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
-        func_80A12C48(this, play, 6);
-        this->unk_3D4++;
+/**
+ * Play the surprise animation then return to shivering.
+ */
+void EnGo_GraveBroGoron_UpdateToSurprised(EnGo* this, PlayState* play) {
```

Comment:
```suggestion
void EnGo_GraveBroGoron_Surprised(EnGo* this, PlayState* play) {
```
For these kind of actions functions, generally just prefer a verb describing what the actor is doing.

Reply:
All of the "UpdateTo" are somewhat of a mix between Setup and Action Function so far as I can tell. They're used primarily within the Setup functions (but occasionally in cutscenes or action functions themselves), to change the animation and action flags that the "true" action function then use.

I was struggling with how to hame these until I saw the example in z_en_s_goro.c and I thought it clicked quite nicely. (Though I now realize I dropped the "Animation" suffix that may have made that more obvious that it's not really an action function.

Would you accept "Engo_GraveBroGoron_UpdateToSurprisedAnimation" (and updating the others accordingly) to keep it in line with the existing z_en_s_goro practice?

Reply:
Yeah that should be fine.